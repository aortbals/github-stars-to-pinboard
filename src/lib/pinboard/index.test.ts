if (process.env.NODE_ENV !== 'production') {
  require('dotenv').config();
}

import { PinboardAPIClient } from '.';

describe('PinboardAPIClient', () => {
  let client: PinboardAPIClient;

  beforeAll(() => {
    client = new PinboardAPIClient({
      authToken: process.env.PINBOARD_API_TOKEN as string,
    });
  });

  describe('#listReposStarredByAuthenticatedUser', () => {
    it('can request a single page with page size', async () => {
      expect(
        await client.add({
          url: 'https://github.com/zeit/async-sema',
          description: 'async-sema',
          extended: 'Semaphore using `async` and `await`',
          dt: '2019-09-21T16:47:07Z',
          tags: [
            'github-starred',
            'async',
            'asynchronous',
            'await',
            'package',
            'semaphore',
          ].join(','),
        })
      ).toMatchSnapshot();
    });
  });
});
