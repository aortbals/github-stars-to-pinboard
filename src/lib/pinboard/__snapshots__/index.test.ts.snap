// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinboardAPIClient #listReposStarredByAuthenticatedUser can request a single page with page size 1`] = `
Object {
  "result_code": "done",
}
`;
